### PR TITLE
Feature/updated documentation executor

### PR DESCRIPTION
**Pull request by JanStaschulat**
updated documentation for number_of_handles in rclc_executor_init function. To clarify that you can add subscriptions at runtime.

### CODE REVIEW

**Comment by JanStaschulat** [comment]
@mergify backport foxy galactic

File: rclc/README.md [hunk 1]
```diff
@@ -273,9 +273,8 @@ Returns a zero initialized executor object.
 **rclc_executor_init(rclc_executor_t * executor, rcl_context_t * context, const size_t number_of_handles, const rcl_allocator_t * allocator)**
 
 As the Executor is intended for embedded controllers, dynamic memory management is crucial.
-Therefore at initialization of the RCLC-Executor, the user defines the total number of handles `number_of_handles`.
-The necessary dynamic memory will be allocated only in this phase and no more memory in the running phase.
-This makes this Executor static in the sense, that during runtime no additional callbacks can be added.
+Therefore at initialization of the RCLC-Executor, the user defines the total number of handles `number_of_handles`. A handle is a term for subscriptions, timers, services, clients and guard conditions. The necessary dynamic memory will be allocated only in this phase and no more memory in the running phase. The corresponding wait-set is allocated in the first execution of the spin-method.
```
Comment (pablogs9) [inline rclc/README.md:276]:
```suggestion
Therefore at initialization of the RCLC-Executor, the user defines the total number of handles `number_of_handles`. A handle is a term for subscriptions, timers, services, clients and guard conditions. The necessary dynamic memory will be allocated only in this phase and no more memory in the running phase. The corresponding wait-set is allocated in the first execution of the spin-method or in the optional call to `rclc_executor_prepare` .
```

File: rclc/include/rclc/executor.h [hunk 2]
```diff
@@ -94,6 +94,15 @@ rclc_executor_get_zero_initialized_executor(void);
...
  *  \p allocator.
+ * As the Executor is intended for embedded controllers, dynamic memory management is crucial.
+ *  Therefore at initialization of the RCLC-Executor, the user defines the total \p number_of_handles.
+ * A handle is a term for subscriptions, timers, services, clients and guard conditions. The
+ * heap will be allocated only in this phase and no more memory will be allocated in the
+ * running phase in the executor. However, the heap memory of corresponding wait-set is
+ * allocated in the first iteration of a spin-method. This makes this Executor static in
+ * terms of memory allocation, in the sense, that during runtime no heap allocations occur.
+ * You can add, however, at runtime as many handles, e.g. subscriptions, to the executor
+ * until the maximum number of handles is reached.
```
Comment (pablogs9) [inline rclc/include/rclc/executor.h:105]: Mention here optional call to `rclc_executor_prepare`

**Review by pablogs9 — CHANGES_REQUESTED** [review]

File: rclc/include/rclc/executor.h (hunk 2 above)
Comment (JanStaschulat) [reply rclc/include/rclc/executor.h]: Resolved in 2f15f80

**Pull request merged**